node/src/components/linear_chain.rs

```diff
@@ -407,9 +427,20 @@ where
                     era_id,
                     ..
                 } = *fs;
-                if let Err(err) = fs.verify() {
-                    warn!(%block_hash, %public_key, %err, "received invalid finality signature");
-                    return Effects::new();
+                if let Some(last_block) = self.latest_block.as_ref() {
+                    let last_block_era = last_block.header().era_id();
+                    let lowest_acceptable_era_id =
+                        last_block_era + self.auction_delay - self.unbonding_delay;
```

Comment:
Actually, we should make a distinction here: If it's a switch block it has already forgotten its own era's validators, unbonded some old validators, and determined new ones. In that case, we should add 1 to `last_block_era`.
```suggestion
                    let current_era = last_block.header().era_id() +
                        if last_block_header.is_switch_block() { 1 } else { 0 };
                    let lowest_acceptable_era_id =
                        current_era + self.auction_delay - self.unbonding_delay;
```

Reply:
Thanks. Fixed in [`31bccd8` (#1144)](https://github.com/CasperLabs/casper-node/pull/1144/commits/31bccd8eacaf40767367af900ea6e25019463f0f).